core/store/src/columns.rs

```diff
@@ -237,6 +237,13 @@ pub enum DBCol {
 }
 
 impl DBCol {
+    /// Whethere this column can be written to only once.
+    ///
+    /// Data in such columns is never overwriten, though it can be deleted by gc
+    /// eventually.
+    pub fn is_write_once(&self) -> bool {
+        WRITE_ONCE_COLUMNS[*self as usize]
+    }
```

Comment:
As an alternative of moving aggregator to misc column, perhaps `is_write_once(&self, key: &[u8]) -> bool { /* ... */ }` and then you can have a special case for AGGREGATOR_KEY?